agent/assets/installconfig/install-agent-config-playbook.yaml

```diff
@@ -0,0 +1,53 @@
+- name: Create install-config and agent-config for Agent Installer
+  hosts: localhost
+  collections:
+   - community.general
+  gather_facts: no
+  vars:
+    - agent_deploy_mce: "{{ lookup('env', 'AGENT_DEPLOY_MCE') }}"
+    - cluster_name: "{{ lookup('env', 'CLUSTER_NAME') }}"
+    - cluster_namespace: "{{ lookup('env', 'CLUSTER_NAMESPACE') }}"
+    - base_domain: "{{ lookup('env', 'BASE_DOMAIN') }}"
+    - num_workers: "{{ lookup('env', 'NUM_WORKERS') }}"
+    - num_masters: "{{ lookup('env', 'NUM_MASTERS') }}"
+    - network_type: "{{ lookup('env', 'NETWORK_TYPE') }}"
+    - ip_stack: "{{ lookup('env', 'IP_STACK') }}"
+    - api_vips: "{{ lookup('env', 'API_VIP') }}"
+    - ingress_vips: "{{ lookup('env', 'INGRESS_VIP') }}"
+    - agent_nodes_macs: "{{ lookup('env', 'AGENT_NODES_MACS_STR') }}"
+    - agent_nodes_ips: "{{ lookup('env', 'AGENT_NODES_IPS_STR') }}"
+    - agent_nodes_ipsv6: "{{ lookup('env', 'AGENT_NODES_IPSV6_STR') }}"
+    - agent_nodes_hostnames: "{{ lookup('env', 'AGENT_NODES_HOSTNAMES_STR') }}"
+    - cluster_host_prefix_v4: "{{ lookup('env', 'CLUSTER_HOST_PREFIX_V4') }}"
+    - cluster_host_prefix_v6: "{{ lookup('env', 'CLUSTER_HOST_PREFIX_V6') }}"
+    - cluster_subnet_v4: "{{ lookup('env', 'CLUSTER_SUBNET_V4') }}"
+    - cluster_subnet_v6: "{{ lookup('env', 'CLUSTER_SUBNET_V6') }}"
+    - service_subnet_v4: "{{ lookup('env', 'SERVICE_SUBNET_V4') }}"
+    - service_subnet_v6: "{{ lookup('env', 'SERVICE_SUBNET_V6') }}"
+    - external_subnet_v4: "{{ lookup('env', 'EXTERNAL_SUBNET_V4') }}"
+    - external_subnet_v6: "{{ lookup('env', 'EXTERNAL_SUBNET_V6') }}"
+    - provisioning_host_external_ip: "{{ lookup('env', 'PROVISIONING_HOST_EXTERNAL_IP') }}"
+    - networking_mode: "{{ lookup('env', 'NETWORKING_MODE') }}"
+    - ssh_pub_key: "{{ lookup('env', 'SSH_PUB_KEY') }}"
+    - pull_secret: "{{ lookup('env', 'PULL_SECRET_FILE') }}"
+    - pull_secret_contents: "{{ lookup('file', pull_secret) | to_json }}"
+    - mirror_images: "{{ lookup('env', 'MIRROR_IMAGES') }}"
+    - mirror_path: "{{ lookup('env', 'MIRROR_PATH') }}"
+    - mirror_command: "{{ lookup('env', 'MIRROR_COMMAND') }}"
+    - local_image_url_suffix: "{{ lookup('env', 'LOCAL_IMAGE_URL_SUFFIX') }}"
+    - local_registry_dns_name: "{{ lookup('env', 'LOCAL_REGISTRY_DNS_NAME') }}"
+    - local_registry_port: "{{ lookup('env', 'LOCAL_REGISTRY_PORT') }}"
```

Comment:
Given that as a first step to replace the bash config files generation with an ansible playbook this approach looks fine, I'm wondering if we could reduce the number of input vars required and "push" some of them within the playbook (like for example the `AGENT_NODES_IPS` or, more in general, some of the logic within `get_static_ips_and_macs`).

Reply:
Yes I'd like to clean up the vars but now these are still used in 04_agent_configure.sh to set up the ZTP manifests. The next step will be to move the ZTP manifest setup to ansible also, at which point we need to make these settings common, via a vars/main.yaml file. So I'd like to push that off to phase2.

Reply:
I agree that we should look at cleaning up the variables in a follow up PR. Maybe once we have more of these files moved into ansible and out of the bash script.